x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/CategoryDefinition.java

```diff
@@ -152,6 +167,26 @@ public void setGrokPattern(String grokPattern) {
         this.grokPattern = grokPattern;
     }
 
+    public long[] getPreferredToCategories() {
+        return preferredToCategories;
```

Comment:
Is it ok to pass array reference here? Should we make a copy maybe?

Reply:
I don't think there is much danger. No code paths are using this getter outside of tests.